trytond/trytond/security.py

```diff
@@ -15,6 +16,20 @@ def _get_pool(dbname):
     database_list = Pool.database_list()
     pool = Pool(dbname)
     if dbname not in database_list:
+        # This solves the following case.
+        #
+        # When the server starts, it will automatically load the pool for all
+        # configured databases, and start the associated cache listeners.
+        #
+        # However, if a user tries to connect to a database that was not
+        # preloaded, the dispatcher has to do all this, right here.
+        # The problem is that the cache listener will be started as soon as the
+        # transaction is created, and its first action will be to stop the pool
+        # / clear the caches.
```

Comment:
I thought that this concurrency had been fixed by f258162adad could it be the `Transaction.start` from `pre_load_database` from this patch that added back another way to have the issue?